packages/block-library/src/pattern/edit.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	store as blockEditorStore,
+	InnerBlocks,
+} from '@wordpress/block-editor';
+
+const PatternEdit = ( { attributes, clientId } ) => {
+	const selectedPattern = useSelect( ( select ) =>
+		select( blockEditorStore ).__experimentalGetParsedPattern(
+			attributes.slug
+		)
+	);
+	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
+	replaceInnerBlocks( clientId, selectedPattern.blocks );
```

Comment:
This is a side effect of the component, so it should be encapsulated and not run inline on any re-render. This could be as simple as using `useEffect` or `useRefEffect`. But the lifecycle of the effect needs to be pondered: when should this be called?, is the effect idempotent?